RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT.cc

```diff
@@ -205,25 +205,28 @@ bool DAClusterizerInZT::merge(vector<vertex_t> & y, double & beta)const{
 }
 
 bool DAClusterizerInZT::purge(vector<vertex_t> & y, vector<track_t> & tks, double & rho0, const double beta)const{
+  constexpr int maxUnique = 2;
   // eliminate clusters with only one significant/unique track
   if(y.size()<2)  return false;
   
   unsigned int nt=tks.size();
   double sumpmin=nt;
   vector<vertex_t>::iterator k0=y.end();
-  for(vector<vertex_t>::iterator k=y.begin(); k!=y.end(); k++){ 
+  const double rho0exp = rho0*std::exp(-beta*dzCutOff_*dzCutOff_);
+  for(vector<vertex_t>::iterator k=y.begin(); k!=y.end(); ++k){ 
     int nUnique=0;
     double sump=0;
-    double pmax=k->pk/(k->pk+rho0*exp(-beta*dzCutOff_*dzCutOff_));
-    for(unsigned int i=0; i<nt; i++){
-      if(tks[i].zi > 0){
-	double p = k->pk * std::exp(-beta*e_ik(tks[i],*k)) / tks[i].zi ;
+    double pmax=k->pk/(k->pk+rho0exp);
+    for(auto&& tk : tks){
+      if(tk.zi > 0){
+	double p = k->pk * std::exp(-beta*e_ik(tk,*k)) / tk.zi ;
 	sump+=p;
-	if( (p > 0.9*pmax) && (tks[i].pi>0) ){ nUnique++; }
+	if( (p > 0.9*pmax) && (tk.pi>0) ){ ++nUnique; }
+	if( nUnique >= maxUnique ) { break; }
```

Comment:
other than this line, I don't see anything obvious that can generate differences in reconstruction.